src/dmd/traits.d

```diff
@@ -2069,7 +2070,42 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto tup = new TupleExp(e.loc, exps);
         return tup.expressionSemantic(sc);
     }
-
+    //https://issues.dlang.org/show_bug.cgi?id=22291
+    if(e.ident == Id.arguments)
+    {
+        //No args are valid
+        if(e.args)
+        {
+            char[] contents = cast(char[]) e.args.toString();
+            contents = contents[1..$];
+            contents[$-1] = '\0';
+            e.error("`__traits(arguments)` cannot have arguments, but `%s` was supplied", contents.ptr);
+            return ErrorExp.get();
+        }
+        if(sc.func is null)
```

Comment:
I suggest adding a newline before this if.  Having clearly delimited paragraphs helps.

Reply:
nit: add space after `if`